### PR TITLE
Kwannoel/summer 2020 blogposts

### PR DESCRIPTION
**Pull request by kwannoel**
To view the blogposts locally: https://github.com/Holmusk/holmusk.github.io#how-to-build-and-run

### CODE REVIEW

File: blog/2020-09-17-Experiences-of-our-engineer-intern-team-Summer-2020.md [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+
+Our team of four software engineer interns has been working on such a solution in the past 2 months, and the product is the **Holmusk Collective**.
+
+Neo4j, a graph database, was used to power our web application. We want to elegantly and efficiently represent Holmusk's resources as a network. To do so, we model it as a [graph](https://en.wikipedia.org/wiki/Graph_(discrete_mathematics)), storing relationships between our resources. 
+
+The diagram below can give you a sneak peek of what is behind the curtains of our solution. 
+
+![](/images/blogposts/neo4j_schema.png)
+
+With reference to the above diagram, this allows our users about relationships. For example, we could ask who has worked on a particular project or with a potential client.
```
Comment (jhrcek) [inline blog/2020-09-17-Experiences-of-our-engineer-intern-team-Summer-2020.md]: This sentence seems to be missing a verb
```suggestion
With reference to the above diagram, this allows our users to learn about relationships. For example, we could ask who has worked on a particular project or with a potential client.
```

File: blog/2020-09-17-Experiences-of-our-engineer-intern-team-Summer-2020.md [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+
+### Database
+
+In order to construct the network that we had in mind, the most prudent choice was to use a graph database (since [networks can be represented as graphs](https://en.wikipedia.org/wiki/Network_theory)). 
+
+We considered [AWS Neptune](https://aws.amazon.com/neptune/) along with other alternatives, but in the end [Neo4j](https://neo4j.com/) was decided on given its ease of use and developer friendliness.
+
+### On the simplicity of Neo4j
+
+The queries in Neo4j pattern match against the actual database structure, which is helpful in gaining intuition in using them. The two most basic components are nodes and relationships.  Using those components, we have the ability to create nodes and relationships between them. The example below can give you some insight about how it works.
```
Comment (jhrcek) [inline blog/2020-09-17-Experiences-of-our-engineer-intern-team-Summer-2020.md]: > The two most basic components are nodes and relationships.  Using those components, we have the ability to create nodes and relationships between them.

These two sentences don't make sense together.
Using those components (nodes and relationships) we have the ability to create nodes and relationships... what? :laughing:

File: blog/2020-09-17-Experiences-of-our-engineer-intern-team-Summer-2020.md [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+
+Let us take a closer look at what that involved.
+
+---
+
+## Functional vs Imperative
+
+In the context of React, there are various types of components. Of interest to us are the *stateless (functional)* and *stateful (imperative) components.*
+
+*Statelessness* means that components to not manipulate any state within, they only render state passed to them in their *props*.
```
Comment (jhrcek) [inline blog/2020-09-17-Experiences-of-our-engineer-intern-team-Summer-2020.md]:
```suggestion
*Statelessness* means that components do not manipulate any state within, they only render state passed to them in their *props*.
```

File: blog/2020-09-17-Experiences-of-our-engineer-intern-team-Summer-2020.md [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+```jsx
+function reducer(state, action) {
+  switch (action.type) {
+    case action1:
+			// do something
+
+    case action2:
+			// do something
+		
+	  case action3:
```
Comment (jhrcek) [inline blog/2020-09-17-Experiences-of-our-engineer-intern-team-Summer-2020.md]: Aesthetic nitpick: Please align this branch with others.

**Review by jhrcek — COMMENTED** [review]
Looks good, just few minor issues.

**Comment by turboMaCk** [comment]
neo4j_db_schema.png  and neo4j_schema.png are the same file

**Review by jhrcek — APPROVED** [review]
Thank you, looks good to me :+1:

**Pull request merged**